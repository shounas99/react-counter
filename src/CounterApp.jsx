import { useState } from 'react'
import PropTypes from "prop-types"
// import confetti from 'https://cdn.skypack.dev/canvas-confetti';

export const CounterApp = ({ value }) => {
  const [ counter , setCounter ] = useState( value );
  const handleAdd = () => { 
    setCounter( counter + 1 )
    if ((counter + 1)% 10 === 0) {
      // confetti();
    }
  }
  const handleSubtract = () => {
    setCounter( counter - 1)
    if ((counter - 1) % 10 === 0) {
      // confetti();
    }
    if(counter - 1 === 0){
      handleReset()
    }
  }
  const handleReset = () => setCounter(value)

  return (
    <>
      <h1>Counter App</h1>
      <h2>{ counter }</h2>

      <button onClick={ handleAdd }> +1 </button>
      <button onClick={ handleSubtract }> -1 </button>
      <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}