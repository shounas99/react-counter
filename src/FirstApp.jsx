//shortcut - rafc 
const newMessage = {
    title: 'Hi',
    message: 'Hola mundo',
}
const getResult = () => {
    return 1 + 1
}


export const FirstApp = ({ title = 'Title', subtitle2 = "second" }) => { /*can I either require a prop to be passed or provide a default value*/
    // we could add constants, variables, or functions within functional components. However, when 
    // a component re-renders, its memory space is re-initialized. Therefore, we need to avoid 
    // writing them directly inside the component's body, even if we need to handle button clicks,
    // execute functions, or manage state changes, all of which may require those constants, variables, or functions.



  return (
    <>
    <h1>{ title }</h1>
    <h1>{ subtitle2 }</h1>
    <h1> { getResult() }</h1>
        <code>{ JSON.stringify(newMessage)  }</code>
        <p>Im subtitle</p>
    </>
  )
}
