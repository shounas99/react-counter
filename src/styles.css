html, body {
  background-color: #21232A;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
  height: 100%;
}

html {
  display: table;
  margin: auto;
}

body {
  display: table-cell;
  vertical-align: middle;
}

button {
  font-size: 1.3rem;
  margin-right: 5px;
  padding: 15px;
  -webkit-user-select: none;
  background: #5E5DF0;
  border-radius: 999px;
  border: 0;
  box-shadow: #5E5DF0 0 10px 20px -10px;
  box-sizing: border-box;
  color: #FFFFFF;
  cursor: pointer;
  font-family: Inter,Helvetica,"Apple Color Emoji","Segoe UI Emoji",NotoColorEmoji,"Noto Color Emoji","Segoe UI Symbol","Android Emoji",EmojiSymbols,-apple-system,system-ui,"Segoe UI",Roboto,"Helvetica Neue","Noto Sans",sans-serif;
  font-size: 16px;
  font-weight: 700;
  line-height: 24px;
  opacity: 1;
  outline: 0 solid transparent;
  padding: 8px 18px;
  touch-action: manipulation;
  user-select: none;
  width: fit-content;
  word-break: break-word;
}

/* button {
    padding: 15px;
    font-size: 1.3rem;
    color: green;
    margin-right: 5px;

    outline: 0;
    grid-gap: 8px;
    align-items: center;
    background-color: #ff90e8;
    color: #000;
    border: 1px solid #000;
    border-radius: 4px;
    cursor: pointer;
    display: inline-flex;
    flex-shrink: 0;
    font-size: 16px;
    gap: 8px;
    justify-content: center;
    line-height: 1.5;
    overflow: hidden;
    padding: 12px 16px;
    text-decoration: none;
    text-overflow: ellipsis;
    transition: all .14s ease-out;
    white-space: nowrap;
}



:hover {
    box-shadow: 4px 4px 0 #000;
    transform: translate(-4px,-4px);
}
:focus-visible{
    outline-offset: 1px;
} */



/* 
body{
    text-align:center;
    background-color:#ffcc8e;
  }
  
  .button{
    position:relative;
    display:inline-block;
    margin:20px;
  }
  
  .button a{
    color:white;
    font-family:Helvetica, sans-serif;
    font-weight:bold;
    font-size:36px;
    text-align: center;
    text-decoration:none;
    background-color:#FFA12B;
    display:block;
    position:relative;
    padding:20px 40px;
    
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
    text-shadow: 0px 1px 0px #000;
    filter: dropshadow(color=#000, offx=0px, offy=1px);
    
    -webkit-box-shadow:inset 0 1px 0 #FFE5C4, 0 10px 0 #915100;
    -moz-box-shadow:inset 0 1px 0 #FFE5C4, 0 10px 0 #915100;
    box-shadow:inset 0 1px 0 #FFE5C4, 0 10px 0 #915100;
    
    -webkit-border-radius: 5px;
    -moz-border-radius: 5px;
    border-radius: 5px;
  }
  
  .button a:active{
    top:10px;
    background-color:#F78900;
    
    -webkit-box-shadow:inset 0 1px 0 #FFE5C4, inset 0 -3px 0 #915100;
    -moz-box-shadow:inset 0 1px 0 #FFE5C4, inset 0 -3pxpx 0 #915100;
    box-shadow:inset 0 1px 0 #FFE5C4, inset 0 -3px 0 #915100;
  }
  
  .button:after{
    content:"";
    height:100%;
    width:100%;
    padding:4px;
    position: absolute;
    bottom:-15px;
    left:-4px;
    z-index:-1;
    background-color:#2B1800;
    -webkit-border-radius: 5px;
    -moz-border-radius: 5px;
    border-radius: 5px;
  }
  
   */