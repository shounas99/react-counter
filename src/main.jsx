import { StrictMode } from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <StrictMode>
    {/* <FirstApp title="Hi! I'm Cris"/> */}
    <CounterApp value={20} />
  </StrictMode>
);
