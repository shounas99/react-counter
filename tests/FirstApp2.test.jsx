import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => { 
    const title = "Hola Joe"
    const subtitle = "Soy un subtitulo"

    test('Debe de hacer match con el snapshot', () => { 
        const { container } = render(
            <FirstApp 
                title = { title }
            />
        )
        expect( container ).toMatchSnapshot()
    })

    test('Debe de mostrar el mensaje "Hola Joe"', () => { 
        render( <FirstApp  title={ title }/>)
        expect( screen.getByText( title )).toBeTruthy() // screen - limpiezas automaticas
        screen.debug()
    })
    
    test('Debe mostrar el titulo en un h1', () => { 
        render( <FirstApp title={ title }/>)
        expect( screen.getByRole("heading", { level: 1 }).innerHTML).toContain( title );
    })
    
    
    test('Debe de mostrar el subtitulo enviado por props', () => { 
        render(
            <FirstApp 
            title={ title }
            subtitle={ subtitle }
            />
        )
        expect( screen.getAllByText( subtitle ).length).toBe(2)
    })

})