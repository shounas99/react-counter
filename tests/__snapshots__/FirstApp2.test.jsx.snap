// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Joe
  </h1>
  <p>
    without subtitle
  </p>
  <p>
    without subtitle
  </p>
  <p />
</div>
`;
